modules/aws/core/vpc.tf

```diff
@@ -277,3 +439,179 @@ resource "aws_route" "peering_accepter" {
   destination_cidr_block    = each.value.destination_cidr_block
   vpc_peering_connection_id = each.value.vpc_peering_connection_id
 }
+
+
+
+# Allow all from eks1-nodes and eks2-nodes subnets to enable outbound
+# internet access from PODs trough NAT Gateway
+# TODO: Do we need anything else in addition to this?
+resource "aws_network_acl" "public" {
+  for_each = {
+    for ix, subnet in concat(local.eks1_nodes_subnets, local.eks2_nodes_subnets) :
+    "${local.public_subnet_name_prefix}-${ix}" => subnet.cidr_block
+  }
+  vpc_id = aws_vpc.this.id
+  subnet_ids = [
+    for subnet in aws_subnet.public :
+    subnet.id
+  ]
+  ingress {
+    protocol   = "-1"
+    rule_no    = 100
+    action     = "allow"
+    cidr_block = each.value
+    from_port  = 0
+    to_port    = 0
+  }
+  tags = merge(
+    local.global_tags,
+    {
+      Name = each.key,
+    },
+  )
+}
+
+# Deny all to private subnets. Access to the private subnets need to be explicitly
+# allowed by network ACLs defined outside this module overriding the below ACL
+resource "aws_network_acl" "private" {
+  for_each = {
+    for ix, subnet in concat(local.eks1_nodes_subnets, local.eks2_nodes_subnets) :
+    "${local.private_subnet_name_prefix}-${ix}" => subnet.cidr_block
+  }
+  vpc_id = aws_vpc.this.id
+  subnet_ids = [
+    for subnet in aws_subnet.private :
+    subnet.id
+    if contains(local.private_subnets.*.cidr_block, subnet.cidr_block)
+  ]
+  ingress {
+    protocol   = "-1"
+    rule_no    = 1000
+    action     = "deny"
+    cidr_block = each.value
+    from_port  = 0
+    to_port    = 0
+  }
+
+  tags = merge(
+    local.global_tags,
+    {
+      Name = each.key,
+    },
+  )
+}
+
+# Allow all from private, public and eks1-cluster subnets 
```

Comment:
I'm probably messing things up but don't you use allow local.private_subnets that you just deny?
Or is this to make it possible to talk from the public network to the private?
If so can we clarify the comment a bit.

Reply:
I think that the fact that we have a subnet named "private" and a number of subnets of type "private" (eks*_nodes, eks*_cluster, private) is messing things up a bit. At least it did for me :-) This ACL is allowing traffic from the subnets named "private", "public" and "eks1_cluster" to "eks1_nodes". I think I should add more documentation to this module visualizing the network setup and the allowed traffic flow between the subnets